babel-boilerplate backbone.radio

paulfalgout opened the pull request.

paulfalgout commented:
This is a WIP

It will resolve #188 

This PR is not converting anything to ES6, just getting it ES6 ready.

I did have to relax a couple of the jscs rules and one jshint rule from the babel-boilerplate setup in order to get the current code to build without errors.

I'm currently having two issues.
1. Travis isn't loading the lodash matrix @megawac ?
2. Again with travis.. I want the code coverage to only run once.. I realize how I have it setup now it would still install codeclimate each time, but I can't seem to get it to work at all anyhow.  Thoughts?

paulfalgout commented:
Ok this is working now.  But there are actually some issues with lodash 3+ in the tests.

jmeas commented:
Lodash support should not be a blocker for this PR. Pull it out of the Travis file, and create a new issue?

paulfalgout commented:
sounds like a plan.  I removed lodash 3+

And I'll open a PR after this with them re-added so there are failing specs.

The only potentially outstanding issue then may be codeclimate.  I'm not entirely sure how to test if code coverage will be working.  I added the token and it's showing up in travis-ci, but I guess it only looks at the master branch (or it is not working)?  Should I have pulled this off master?

jasonLaster commented:
damn, this is a lot, but i like it. good job w the needed migrations. and staying true to the babel-boilerplate

I'm not sure how to asses code-climate. I'm okay, merging and then fixing if it's an issue...

I agree that lodash doesn't need to be here.

paulfalgout commented:
I think this is ready to go.. just need :eyes: and :+1: s

jasonLaster commented:
Great work Paul! This is a huge step forward for us

jmeas commented:
This should have just gone into master
